src/ipld/util.rs

```diff
@@ -401,8 +407,7 @@ impl<DB: Blockstore + Send + Sync + 'static, T: Borrow<Tipset>, ITER: Iterator<I
     ) -> JoinHandle<anyhow::Result<()>> {
         task::spawn(async move {
             let mut handles = JoinSet::new();
-
-            for _ in 0..num_cpus::get() {
+            for _ in 0..num_cpus::get().clamp(1, 8) {
```

Comment:
Do you think it makes sense to accept this as params as well?

Basically allow user to pass the worker count otherwise use current values as default?

Reply:
I think it's our responsibility to tune this value. It might turn out this `unordered_stream_chain` is not fast and we end up removing it.